### ISSUE TITLE
Question: Pub/Sub state in the SpaceFOMRefFrameSimObject with trick variables

### ISSUE THREAD

**Issue by simtheverse** [nasa/TrickHLA#127]
Hello,
I am trying to use trick variables to both drive the state data in a SpaceFOMRefFrameSimObject as well as have other trick variables be driven by state data in a SpaceFOMRefFrameSimObject. 

It looks like SpaceFOMRefFrameObject has 3 attributes, and one of them is 'state'. The trick_name in the state attribute is `ref_frame_Chaser.frame_packing.stc_encoder.buffer`.  How do I tie trick variables to this buffer to both publish and subscribe? Is there a simple example by chance?

Thanks for the help!

**Comment by ezcrues** [comment]
Yeah, that stuff is broken right now.  I am working with Dan to create a version that will work for lag compensation and ownership transfer.  Unfortunately, we are probably going to have to change some core TrickHLA to fix this right.  We've actually be working on this for a while.

We'll up date this thread as we make progress.  I'm hoping to have something that works soon.

**Comment by ezcrues** [comment]
I am still working on this but I think I have an example that works.  You can check out the jeod_examples branch and look at sims/SpaceFOM/SIM_Entity_Test.  I've refactored the SpaceFOM base classes to support a method that may give you what you are looking for.  I've only done a little testing but it seems to work for all cases of lag compensation: None, Sending, and Receive.  Once I clean this up, I will work up a JEOD example using this new approach.
